sklearn/impute/_base.py

```diff
@@ -667,7 +686,13 @@ def fit_transform(self, X, y=None):
             will be boolean.
 
         """
-        return self.fit(X, y).transform(X)
+        imputer_mask = self._fit(X, y)
+
+        if (self.features == "missing-only") and \
+                (self.features_.size < self._n_features):
```

Comment:
I think this is a sufficient condition. Remove the other.

Reply:
You mean `(self.features_.size < self._n_features)`, this one?

Reply:
I mean remove `(self.features == "missing-only")`

Reply:
But what if self.features is 'all' bcs in _fit we are ensuring that the features should be either 'missing-only' or 'all'. So, this particular block of code will execute even when the value of self.features is 'all'.
Is this acceptable?? will it not create any further problem??

Thanks.

Reply:
If features='all', then `self.features_.size <  self._n_features` can not be true.

"mask" is, to my understanding, taken from painting where the mask covers some parts of the canvas and uncovers others. It's used in numeric computing to mean a Boolean array used to select cells from an array of the same size.

Reply:
> If features='all', then `self.features_.size < self._n_features` can not be true.

oh yeah, sorry, for that silly thing.

I'll make the necessary changes right now.

Thanks